pkg/controllers/state/cluster.go

```diff
@@ -90,6 +104,14 @@ func (c *Cluster) Synced(ctx context.Context) (synced bool) {
 	// Set the metric to whatever the result of the Synced() call is
 	defer func() {
 		clusterStateSynced.Set(lo.Ternary[float64](synced, 1, 0))
+		if synced {
+			c.lastSynced = time.Now()
+		}
+		// If we have been out of sync for a while, and we haven't logged about it recently, log it
+		if time.Since(c.lastSynced) > AbnormalClusterSyncDuration && c.cm.HasChanged("ClusterSynced", synced) {
```

Comment:
I'm still surprised that this works with the existing code. I'd imagine that you'd need to set `c.lastSynced` to a value right when we startup; else, you are going to get it set to the zero value if it doesn't sync and it's going to result in the log that you have been seeing where we are reporting that we haven't synced for a _very_ long time :D

Reply:
Yep that makes sense the first synced log would be from 292 years ago. Modified it to set last synced to time.Now() on startup.